gulpfile.js

```diff
@@ -33,13 +37,62 @@ const FONTS_DIR = 'build/fonts';
 const IMAGES_DIR = 'build/images';
 const TEMPLATES_DIR = 'src/sidebar/templates';
 
+const webtranslateit = 'https://webtranslateit.com/api/projects/';
+const webtranslateitReadKey = 'proj_pub_h-TswYPuN3K7EnSlMwcEdw';
+const captionFiles = {
+    prodFile: '734937',
+    locales: ['en-US', 'nb-NO', 'nn-NO', 'sv-SE', 'pl-PL', 'es-CO' ],
+};
+
 // LiveReloadServer instance for sending messages to connected
 // development clients
 let liveReloadServer;
 // List of file paths that changed since the last live-reload
 // notification was dispatched
 let liveReloadChangedFiles = [];
 
+function webTranslateFile(key, file, locale) {
+  return {
+      uri: `${webtranslateit}${key}/files/${file}/locales/${locale}`,
+      json: true,
+  };
+}
+function fetchFile(source) {
+  return request(source, (error) => {
+    if (error !== null) {
+      process.stdout.write(error);
+    }
+  });
+}
+
+function fetchCaptions(data) {
+  const file = data.prodFile;
+  const promises = [];
+  const json = {};
+  if (!fs.existsSync('i18n')) {
+    mkdirp('i18n');
+  }
+  for (let i = 0; i < data.locales.length; i++) {
+    const locale = data.locales[i];
+    promises.push(fetchFile(webTranslateFile(webtranslateitReadKey, file, locale)));
```

Comment:
My suggestion:
```suggestion
    function fetchWebTranslateFile(webtranslateitReadKey, file, locale) {
        return fetchFile(webTranslateFile(webtranslateitReadKey, file, locale)).then((captions) => ({ locale, captions }));
    }
```

Then you do not need the Object.keys(json) looping below.

Reply:
always improving things a lot, going to take note on your use of reducer method, thanks a lot